tests/integration_tests/core_tests.py

```diff
@@ -603,6 +608,26 @@ def test_warm_up_cache(self):
             + quote(json.dumps([{"col": "name", "op": "in", "val": ["Jennifer"]}]))
         ) == [{"slice_id": slc.id, "viz_error": None, "viz_status": "success"}]
 
+    @pytest.mark.usefixtures("load_birth_names_dashboard_with_slices")
+    @mock.patch.object(ChartDataCommand, "run")
+    def test_warm_up_cache_error(self, run_mock) -> None:
+        self.login()
+        slc = self.get_slice("Pivot Table v2", db.session)
```

Comment:
Another sad byproduct of [having multiple ways of doing things](https://github.com/apache/superset/blob/95db6c04aaec388210a30eed70e927ebe2484f9c/superset/views/core.py#L1749-L1775). The `Empty query?` is handled differently depending on whether or not the slice has a query context defined as it seems other paths wont raise, i.e., the error is stringified via the `str` function.